### DISCUSSION TITLE
How to fire js event from alpine component Js

### DISCUSSION THREAD

**Discussion by lutfifirdaus** [filamentphp/filament#14057] [Help]
### Package

Other

### Package Version

v3.2

### How can we help you?

Hello guys. I wanna try creating qrscanner wrapper and im using [alpine component](https://filamentphp.com/docs/3.x/support/assets#registering-javascript-files) to export the asset, when on succes i wanna pass the variable using firing event listener to livewire component but when i try dispatching, none result. Anyone know how I can pass the data from alpine component

![image](https://github.com/user-attachments/assets/9c24491f-19b4-4feb-bd69-a584669c64af)
![image](https://github.com/user-attachments/assets/77f87518-265b-47d8-a129-3ac710aa3f27)

Edit :
Here is the code @leandrocfe 

I'm using this [html5-qrcode](https://github.com/mebjas/html5-qrcode) for my scanner

on `AppServiceProvider`
```
 FilamentAsset::register([
            AlpineComponent::make('qr-scanner', __DIR__ . '/../../resources/js/dist/plugin/qr-scanner.js'),
           ]);
```

on `qr-scanner.js`
```
import { Html5QrcodeScanner } from "html5-qrcode";

export default function qrScanner(args) {

    return {
        qrScanner: null,
        id: args.id,
        init() {
            this.qrScanner = new Html5QrcodeScanner(
                this.id,
                { fps: 10, qrbox: { width: 250, height: 250 } },
                /* verbose= */ false
            );

            this.qrScanner.render(this.onScanSuccess, this.onScanFailure);
        },
        onScanSuccess(decodedText, decodedResult) {
            const event = new CustomEvent('scan-success', {text:decodedText, result:decodedResult})
            document.dispatchEvent(event);
        },
        onScanFailure(error) {
            console.warn(`Code scan error = ${error}`);
        },
    };
}
```

on `QrScanner.php`
```
 #[On('scan-success')]
    public function scanResult($result): void
    {
        dd($result);
    }
```

on blade
```
<div
    ax-load
    ax-load-src="{{ \Filament\Support\Facades\FilamentAsset::getAlpineComponentSrc('qr-scanner') }}"
    x-data="qrScanner({ id: '{{ 'reader' }}' })">
    <div
        id="reader"
        width="600px">
    </div>
</div>
```

**Comment by leandrocfe** [comment]
Could you share the code (not image) please?

**Reply by lutfifirdaus** [reply]
I've added the code sir
